Remove Hyrax::BasicMetadata for AllinsonFlex

orangewolf opened the pull request:
# Summary
Addresses #199 by removing Hyrax::BasicMetadata and modifying AllinsonFlex to temporarily define attributes which cannot be adequately supported as required in Hyrax via dynamic schema.

# Screenshots / Video

# Expected Behavior

# Notes

If you need further information on how to fill this out, read the [playbook](https://playbook-staging.notch8.com/en/notch8/process/merge-requests)

Commit:
Remove Allinson Flex dynamic metadata from audio

Include was moved to SharedWorkBehavior

Commit:
WIP removing Hyrax::BasicMetadta

Replaces with new AllinsonFlex::FoundationalMetadata

Commit:
Merge branch 'main' into remove-basic-metadata

Commit:
Merge branch 'main' into remove-basic-metadata

Commit:
Merge branch 'main' into remove-basic-metadata

Commit:
Merge branch 'main' into remove-basic-metadata

Commit:
Update Allison Flex

aprilrieger committed:
add return to get a commit to run the workflows

add return to get a commit to run the workflows

ShanaLMoore commented on Gemfile.lock:
```diff
@@ -8,7 +8,7 @@ GIT
 GIT
   remote: https://github.com/samvera-labs/allinson_flex.git
-  revision: 2c930adc0e4c88a6c5a44c4f43e5b9a6f41666fe
+  revision: 96b7c48da97698c4d09c63f9b8b1fcbe1f763b49
```
Revert the last two commits
Run:
bundle update allinson_flex --conservative 

We should only see an update to the revision in gemfile.lock

ShanaLMoore replied:
👍🏿

ShanaLMoore reviewed (changes requested):
Nice work. This is working after making a few adjustments...

Locally I reverted the last two commits, updated allinson flex conservatively, corrected a rubocop issue, and needed to remove [this](https://github.com/scientist-softserv/utk-hyku/blob/main/app/indexers/app_indexer.rb#L10) line in order to successfully save any work type. 

Leaving that line in app indexer caused the creation of a work to fail. It is basically looking for all the properties defined in hyrax basic metadata, so since we removed most of them it caused issues. 

ref: https://github.com/scientist-softserv/utk-hyku/pull/168/files

Commit:
Revert "add return to get a commit to run the workflows"

This reverts commit 54e51a3f3eb2075a6a2ae55367866a13f03ef692.

Commit:
Revert "Update Allison Flex"

This reverts commit 5ad7297d27b2dbdf476e7fc8ea196a0c47ad489a.

Commit:
Conservative update Allinson Flex

Commit:
Merge branch 'main' into remove-basic-metadata

Commit:
Remove deep indexing

Deep indexing fails because it requires field 'description' which is not included on UTK works.
This should be handled in AllinsonFlex's 'FoundationalMetadata'.

laritakr commented:
> #### ⚠️ GitGuardian has uncovered 2 secrets following the scan of your pull request.
> Please consider investigating the findings and remediating the incidents. Failure to do so may lead to compromising the associated services or software components.
> 
> 🔎 Detected hardcoded secrets in your pull request
> 🛠 Guidelines to remediate hardcoded secrets
> 🦉 [GitGuardian](https://dashboard.gitguardian.com/auth/login/?utm_medium=checkruns&utm_source=github&utm_campaign=cr1) detects secrets in your source code to help developers and security teams secure the modern development process. You are seeing this because you or someone else with access to this repository has authorized GitGuardian to scan your pull request.Our GitHub checks need improvements? [Share your feedbacks](https://050127dayho.typeform.com/to/KmeAPTMk)!

**These secrets warnings are brought back in due to merging the main branch updates into my branch. Nothing I changed included any secrets.**

ShanaLMoore commented on app/indexers/app_indexer.rb:
```diff
@@ -6,8 +6,8 @@ class AppIndexer < Hyrax::WorkIndexer
-  # You can remove this if you don't want this behavior
-  include Hyrax::IndexesLinkedMetadata
+  # Utk does not include based_near and does not need deep indexing.
+  # include Hyrax::IndexesLinkedMetadata
```
👍🏿

ShanaLMoore commented on app/indexers/attachment_indexer.rb:
```diff
@@ -3,14 +3,6 @@
 # Generated via
 #  `rails generate hyrax:work Attachment`
 class AttachmentIndexer < AppIndexer
-  # This indexes the default metadata. You can remove it if you want to
```
👍🏿
